mock/src/transaction.rs

```diff
@@ -160,6 +162,8 @@ impl Default for MockTransaction {
             max_priority_fee_per_gas: Word::zero(),
             max_fee_per_gas: Word::zero(),
             chain_id: *MOCK_CHAIN_ID,
+            invalid: false,
+            forced_nonce: None,
```

Comment:
It seems to me that new defined `force_nonce` is a bit confused.

Can we just manipulate original `nonce` value by new setter `set_nonce` when we are unittest invalid transaction? ( I believe there must be some obstacle so we need to introduce it)

Reply:
The reason I introduced the new variable is because I didn't want to change the existing `nonce` value, but it turns out it's actually a small change, so just made `nonce` an option. Also renamed to `set_nonce` because it seems people like that name better.